pkg/daemon/ceph/client/mirror.go

```diff
@@ -136,8 +136,8 @@ func enablePoolMirroring(context *clusterd.Context, clusterInfo *ClusterInfo, po
 		return errors.Wrapf(err, "failed to get mirroring info for the pool %q", pool.Name)
 	}
 
-	if pool.Mirroring.Mode == mirrorInfo.Mode {
-		logger.Debugf("mirroring is already enabled on the pool %s with mode %s", pool.Name, mirrorInfo.Mode)
+	if mirrorInfo.Mode != "" && mirrorInfo.Mode != "disabled" && pool.Mirroring.Mode != mirrorInfo.Mode {
```

Comment:
Tangential to the architecture discussions, it seems strange for me for a helper method named `enablePoolMirroring()` (and which has a `DisablePoolMirroring()` peer) to be checking whether the config is empty/disabled. 

Given the name of this helper function, I would expect that the caller checks whether it's safe to enable or not and this helper just implements simple, idempotent "enable" behavior.